### PR TITLE
RFC: Add Model Deployment curriculum

### LINKED ISSUE #37: Missing curriculum for S6 - Data Science in the real world
# Missing curriculum for S6 - Data Science in the real world

Currently, there are no guidelines for the S6 curriculum. These should be added to the repo containing what is currently being covered in the academy. I'm tagging the people who were responsible for S6 on batch5.

### LINKED ISSUE #38: [S06 - BLU14] Add info on how to retrieve data from DB on Heroku
# [S06 - BLU14] Add info on how to retrieve data from DB on Heroku

## Context

Currently, we are providing information on BLU on how to reset the Postgres DB in Heroku, but we never cover how to retrieve the information in the database. This is necessary for data drift analysis and/or redeployment. 

## Detailed Description

This is affecting students, particularly during the Capstone project, as a lot of time is spent trying to retrieve the data which is a fairly straightforward thing to do if you know how to. 

##  Possible implementation

A markdown file can be added to BLU14 including instructions on how to retrieve the data. There are two possible ways:
- Using the credentials and a 3rd party program such as DBeaver (such as the one featured in specialization 2)
- Using the dataclip feature in Heroku

### LINKED ISSUE #39: [S06 - BLU13] - Test database storage
# [S06 - BLU14] - Test database storage

## Context

This is related to [this issue](https://github.com/LDSSA/heroku-model-deploy/issues/19) 

## Detailed Description

A few students have reached the capstone without having set up the database correctly in Heroku and without noticing that no observations were being stored.

Locally, the line in [this issue](https://github.com/LDSSA/heroku-model-deploy/issues/19) works but it doesn't allow Postgres to link to the app.

## Possible Implementation

In BLU13, we are testing the predict/ endpoint. We can follow the same approach by asking to create a "retrieve/" endpoint that responds with the contents of the DB given an ID

### PR DESCRIPTION
**Pull request by UrbanoFonseca**
# Model Deployment Curriculum

- Add missing curriculum for S06
- Reorganized current structure in "BLU15 - Model CSI"
- Added "retrieve/" endpoint to BLU13 to test database storage (cf. https://github.com/LDSSA/curriculum-development/issues/39)
- Added optional "Retrieving data" guide (cf. https://github.com/LDSSA/curriculum-development/issues/38)

## Contributing Guidelines

- [x] I have read the **CONTRIBUTING** document.

## Related Issue

Closes #37 
Closes #38 
Closes #39

### CODE REVIEW

**Comment by UrbanoFonseca** [comment]
The curriculum may also include the suggestions in #38  and #39.

File: curriculum/06-model-deployment.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+3. Uptime and surviving failures
+
+##### Detailed curriculum
+
+1. Formalizing client requirements
+
+    1. Expectation vs reality when dealing with clients
+    2. Converting business requirements to technical metrics
+    3. Fairness & privacy requirements, model prioritization
+
```
Comment (UrbanoFonseca) [inline curriculum/06-model-deployment.md:80]: 4. Technical architecture diagram with database, webserver, endpoints, etc.

This should provide a high level overview of how everything fits together to suit client requirements

Comment (UrbanoFonseca) [reply curriculum/06-model-deployment.md:80]: 5. Threshold tuning cf. business requirements (precision, recall)

**Comment by UrbanoFonseca** [comment]
cc/ @cimendes

**Comment by cimendes** [comment]
@UrbanoFonseca Is this ready to be merged?

**Comment by UrbanoFonseca** [comment]
@cimendes I was waiting for comments first. Since you are the lead for this year's S06, let me know if you agree and I'll merge.

**Review by cimendes — APPROVED** [review]

**Comment by cimendes** [comment]
> @cimendes I was waiting for comments first. Since you are the lead for this year's S06, let me know if you agree and I'll merge.

Just approved the changes. I've nothing to add to the "Topics not covered" so I'm okay with leaving it as "WIP".

**Pull request merged**